src/app/InteractionModelEngine.h

```diff
@@ -124,7 +125,10 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate
      */
     uint16_t GetReadClientArrayIndex(const ReadClient * const apReadClient) const;
 
+    reporting::Engine & GetReportingEngine() { return mReportingEngine; }
+
 private:
+    friend class reporting::Engine;
```

Comment:
why does this need to be a friend?

Generally any friend declarations concern me since it indicates tight coupling.

Reply:
reporting engine need to get readerhandler from mReadHandlers stored in Interaction model engine, reader handler should not be exposed to consumer sdk, so in order to make reader handler visible in reporting engine, i add friend class here. thanks